### DISCUSSION TITLE
Error when using Nextcloud Assistant

### DISCUSSION THREAD

**Discussion by ebildebil** [nextcloud/all-in-one#5360] [Questions] [answered]
Hi,

I have configured an external API for use with Nextcloud Assistant. In this case Mistral.
When I tried a prompt today I get the below error:
My promt has a total of: 1009 characters and 182 words.

```
Text generation error: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 7 ERROR: value too long for type character varying(1000)
```
In the logs I see below error:

```
{"reqId":"V85FoTrYUb0uA2GqxpJj","level":3,"time":"2024-10-01T16:46:29+00:00","remoteAddr":"127.0.0.1","user":"bisu","app":"PHP","method":"POST","url":"/apps/assistant/f/process_prompt","message":"Undefined array key 7 at /var/www/html/lib/private/AppFramework/Http.php#128","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36","version":"29.0.7.1","data":{"app":"PHP"},"id":"66fc27c87b130"}
```

General googling suggests that it is likely a limitation in number of charecters that can be stored in the DB for that field? (I could be wrong)
If so, the newer models, have a context size of 128k, roughly 80k English words. 

I have set 18000 as the prompt token limit.

Is this something easily solvable (perhaps I'm doing something wrong?)

**When I edit my promt to 991 chareters and 178 words, I do not get an error.**

**Comment by szaimen** [comment, marked as answer]
Hi, please report this to https://github.com/nextcloud/assistant/issues. Thanks!
